### PR TITLE
feat(server): update to Hapi 17

### PR DESCRIPTION
**Pull request by vladikoff**

### CODE REVIEW

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: b287fc840aff7b6c555ca3140765a67480a43d1b
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-oauth-server

---
This message was auto-generated by https://gitcop.com

File: lib/server/internal.js [hunk 1]
```diff
@@ -12,17 +12,14 @@ const logger = require('../logging')('server.clients');
...
 const summary = require('../logging/summary');
 
-exports.create = function createServer() {
+exports.create = async function createServer() {
   var isProd = env.isProdLike();
 
-  var server = new Hapi.Server(
-    require('./config')
-  );
+  var serverConfig = require('./config');
```
Comment (vbudhram) [inline lib/server/internal.js]: nit: `const`

File: lib/routes/root.js [hunk 2]
```diff
@@ -31,16 +31,24 @@ module.exports = {
...
-      var configPath = path.join(gitDir, 'config');
-      var cmd = 'git config --get remote.origin.url';
-      exec(cmd, { env: { GIT_CONFIG: configPath, PATH: process.env.PATH }}, function(err, stdout) {
-        source = stdout.replace(/\s+/, '');
-        return sendReply();
+    function runGitCmd() {
+      return new Promise((resolve) => {
+        // figure it out from .git
+        var gitDir = path.resolve(__dirname, '..', '..', '.git');
+        exec('git rev-parse HEAD', { cwd: gitDir }, function(err, stdout) {
```
Comment (vbudhram) [inline lib/routes/root.js]: Should this reject if an error is found?

Comment (vbudhram) [reply lib/routes/root.js]: Might be able to clean up some of callback nesting with http://bluebirdjs.com/docs/api/promise.fromcallback.html, but meh.

File: bin/internal.js [hunk 3]
```diff
@@ -9,8 +9,10 @@ const server = require('../lib/server/internal').create();
 
 logger.debug('config', config);
 db.ping().done(function() {
-  server.start(function() {
-    logger.info('listening', server.info.uri);
+  server.then((s) => {
+    s.start(function() {
```
Comment (vbudhram) [inline bin/internal.js]: If we are changing lines, might be worthwhile to update to newer syntax?

Comment (vladikoff) [reply bin/internal.js]: 👍

File: lib/routes/root.js (hunk 2 above)
Comment (vladikoff) [reply lib/routes/root.js]: I think in this case we just don't report the information for the `source` and etc

**Pull request merged**